src/nci_util.c

```diff
@@ -849,6 +849,105 @@ nci_parse_intf_activated_ntf(
     return FALSE;
 }
 
+static
+gsize nci_mode_param_inplace_copy(const NciModeParam* param, NciModeParam* copy,
+    NCI_MODE mode)
+{
+    if (G_LIKELY(param) && G_LIKELY(copy)) {
+        gsize size = G_ALIGN8(sizeof(NciModeParam));
+        const NciModeParamPollB* poll_b = NULL;
+        const NciModeParamListenF* listen_f = NULL;
+
+        switch (mode) {
+        case NCI_MODE_ACTIVE_POLL_A:        /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_A:       /* fallthrough */
+        case NCI_MODE_ACTIVE_POLL_F:        /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_F:
+            memcpy(copy, param, size);
```

Comment:
I would rather copy `sizeof(NciModeParam)` bytes in this case, not assuming that the caller has allocated aligned number of bytes (although it might actually be the case)